### ISSUE TITLE
Missing CUDA 10.1 and 10.2 packages

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/faiss-split-feedstock#9]
We tried to use the CUDA packages built here recently, but found there were no packages for CUDA 10.1 or 10.2. Though the feedstock is configured to build both [10.1]( https://github.com/conda-forge/faiss-split-feedstock/blob/7ef0a23d5f2384674c8a351398650d19eb1e1e1a/.ci_support/linux_cuda_compiler_version10.1.yaml ) and [10.2]( https://github.com/conda-forge/faiss-split-feedstock/blob/7ef0a23d5f2384674c8a351398650d19eb1e1e1a/.ci_support/linux_cuda_compiler_version10.12.yaml ).

Tried re-rendering with issue ( https://github.com/conda-forge/faiss-split-feedstock/issues/7 ), but the bot generated PR ( https://github.com/conda-forge/faiss-split-feedstock/pull/8 ) doesn't seem to be doing anything other than removing the workaround added in PR ( https://github.com/conda-forge/faiss-split-feedstock/pull/5 ), which we still need due to space limitations. So closed that out.

Tried [rerunning CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=177790&view=results ) ([logs attached]( https://github.com/conda-forge/faiss-split-feedstock/files/4823088/logs_177790.zip
 )). While all the builds passed, nothing appears to have been uploaded. Looking at the CI logs for CUDA 10.1 and 10.2, it appears the token may be missing. Attached a snippet of what happens for the CUDA 10.2 build, but the CUDA 10.1 build appears the same.

<details>

```
+ validate_recipe_outputs faiss-split-feedstock
Cloning into '/tmp/tmpft8kzycv_recipe/feedstock-outputs'...
validation results:
{
  "linux-64/faiss-1.6.3-py37h4315b5e_1_cuda.tar.bz2": true,
  "linux-64/faiss-gpu-1.6.3-py38_1.tar.bz2": true,
  "linux-64/faiss-1.6.3-py38hfc5fb95_1_cuda.tar.bz2": true,
  "linux-64/libfaiss-1.6.3-he61ee18_1_cuda.tar.bz2": true,
  "linux-64/faiss-1.6.3-py36hb679d2d_1_cuda.tar.bz2": true,
  "linux-64/faiss-gpu-1.6.3-py36_1.tar.bz2": true,
  "linux-64/faiss-gpu-1.6.3-py37_1.tar.bz2": true,
  "linux-64/faiss-proc-1.0.0-cuda.tar.bz2": true
}
NOTE: Any outputs marked as False are not allowed for this feedstock.
+ [[ True != \F\a\l\s\e ]]
+ upload_package --validate --feedstock-name=faiss-split-feedstock /home/conda/feedstock_root /home/conda/recipe_root /home/conda/feedstock_root/.ci_support/linux_cuda_compiler_version10.2.yaml
Found git SHA 7ef0a23d5f2384674c8a351398650d19eb1e1e1a for this build!
Using BINSTAR_TOKEN for anaconda.org uploads to cf-staging.
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
Adding in variants from /home/conda/feedstock_root/.ci_support/linux_cuda_compiler_version10.2.yaml
Distribution /home/conda/feedstock_root/build_artifacts/linux-64/faiss-1.6.3-py37h4315b5e_1_cuda.tar.bz2 is new for cf-staging, but no upload is taking place because the BINSTAR_TOKEN/STAGING_BINSTAR_TOKEN is missing or empty.
Distribution /home/conda/feedstock_root/build_artifacts/linux-64/faiss-gpu-1.6.3-py38_1.tar.bz2 is new for cf-staging, but no upload is taking place because the BINSTAR_TOKEN/STAGING_BINSTAR_TOKEN is missing or empty.
Distribution /home/conda/feedstock_root/build_artifacts/linux-64/faiss-1.6.3-py38hfc5fb95_1_cuda.tar.bz2 is new for cf-staging, but no upload is taking place because the BINSTAR_TOKEN/STAGING_BINSTAR_TOKEN is missing or empty.
Distribution /home/conda/feedstock_root/build_artifacts/linux-64/libfaiss-1.6.3-he61ee18_1_cuda.tar.bz2 is new for cf-staging, but no upload is taking place because the BINSTAR_TOKEN/STAGING_BINSTAR_TOKEN is missing or empty.
Distribution /home/conda/feedstock_root/build_artifacts/linux-64/faiss-1.6.3-py36hb679d2d_1_cuda.tar.bz2 is new for cf-staging, but no upload is taking place because the BINSTAR_TOKEN/STAGING_BINSTAR_TOKEN is missing or empty.
Distribution /home/conda/feedstock_root/build_artifacts/linux-64/faiss-gpu-1.6.3-py36_1.tar.bz2 is new for cf-staging, but no upload is taking place because the BINSTAR_TOKEN/STAGING_BINSTAR_TOKEN is missing or empty.
Distribution /home/conda/feedstock_root/build_artifacts/linux-64/faiss-gpu-1.6.3-py37_1.tar.bz2 is new for cf-staging, but no upload is taking place because the BINSTAR_TOKEN/STAGING_BINSTAR_TOKEN is missing or empty.
Distribution /home/conda/feedstock_root/build_artifacts/linux-64/faiss-proc-1.0.0-cuda.tar.bz2 is new for cf-staging, but no upload is taking place because the BINSTAR_TOKEN/STAGING_BINSTAR_TOKEN is missing or empty.
```

</details>

cc @cjnolet @kkraus14

**Comment by jakirkham** [comment]
@beckermr, do you have any thoughts as to why these builds wouldn't have uploaded? Is there something we are missing?

**Comment by h-vetinari** [comment]
That's a pity, sorry. I checked that the CI ran through on master, but didn't check that the build was uploaded...

**Comment by jakirkham** [comment]
No need to be sorry. In fact thanks for all of the work you have already done. This is really great! 😄

Just need to figure out the setting issues here.

**Comment by beckermr** [comment]
The manual edits to the build scripts removed the binstar tokens from the jobs. 

https://github.com/conda-forge/faiss-split-feedstock/blob/master/.azure-pipelines/azure-pipelines-linux.yml#L48

If you need this diskspace management, i'd submit it as a patch to conda smithy.

**Comment by jakirkham** [comment]
Interesting, so why do the manual edits cause this?

**Comment by isuruf** [comment]
If you look at the code in https://github.com/conda-forge/faiss-split-feedstock/pull/8/files, you'll see why.

**Comment by h-vetinari** [comment]
Closing as fixed by #5, see [CI log](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=178734&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14&t=937c195f-508d-5135-dc9f-d4b5730df0f7) from master:
```
+ upload_package --validate --feedstock-name=faiss-split-feedstock /home/conda/feedstock_root /home/conda/recipe_root /home/conda/feedstock_root/.ci_support/linux_cuda_compiler_version10.1.yaml
Found git SHA df19d1cd4e7fde1492854bccd9b8a36a53775e18 for this build!
Using STAGING_BINSTAR_TOKEN for anaconda.org uploads to cf-staging.
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
Adding in variants from /home/conda/feedstock_root/.ci_support/linux_cuda_compiler_version10.1.yaml
copy results:
{
  "errors": [],
  "valid": {
    "linux-64/libfaiss-1.6.3-he68dc02_1_cuda.tar.bz2": true,
    "linux-64/faiss-1.6.3-py38hd55746d_1_cuda.tar.bz2": true,
    "linux-64/faiss-gpu-1.6.3-py38_1.tar.bz2": true,
    "linux-64/faiss-1.6.3-py37hddabd6f_1_cuda.tar.bz2": true,
    "linux-64/faiss-gpu-1.6.3-py36_1.tar.bz2": true,
    "linux-64/faiss-gpu-1.6.3-py37_1.tar.bz2": true,
    "linux-64/faiss-1.6.3-py36h13546fc_1_cuda.tar.bz2": true,
    "linux-64/faiss-proc-1.0.0-cuda.tar.bz2": true
  },
  "copied": {
    "linux-64/libfaiss-1.6.3-he68dc02_1_cuda.tar.bz2": true,
    "linux-64/faiss-1.6.3-py38hd55746d_1_cuda.tar.bz2": true,
    "linux-64/faiss-gpu-1.6.3-py38_1.tar.bz2": true,
    "linux-64/faiss-1.6.3-py37hddabd6f_1_cuda.tar.bz2": true,
    "linux-64/faiss-gpu-1.6.3-py36_1.tar.bz2": true,
    "linux-64/faiss-gpu-1.6.3-py37_1.tar.bz2": true,
    "linux-64/faiss-1.6.3-py36h13546fc_1_cuda.tar.bz2": true,
    "linux-64/faiss-proc-1.0.0-cuda.tar.bz2": true
  }
}
+ touch /home/conda/feedstock_root/build_artifacts/conda-forge-build-done-linux_cuda_compiler_version10.1
+ test -f /home/vsts/work/1/s/build_artifacts/conda-forge-build-done-linux_cuda_compiler_version10.1
```

Same for [10.2](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=178734&view=logs&j=517fe804-fa30-5dc2-1413-330699242c05&t=c7d0fcb4-3a10-5b26-dfea-e879a93fc2ab).
